File: C++/Leetcode/Array/buySellStockII.cpp
#include <iostream>
#include <vector>

class Solution {
    public:
        int maxProfit(std::vector<int>& prices){
            int profit = 0;
            for (int i = 0; i < prices.size()-1; ++i){
                if (prices[i+1]>prices[i]){
                    profit += prices[i+1]-prices[i];
                }
            }
            return profit;
        };
};


int main(){
    Solution solution_instance;
    std::vector<int> nums = {7,1,5,3,6,4};
    std::cout << solution_instance.maxProfit(nums) << std::endl;
    return 0;
}

File: Python/Leetcode/Array/buySellStockII.py
class Solution(object):
    def maxProfit(self, prices):
        total = 0
        for i in range(len(prices)-1):
            if prices[i+1] > prices[i]:
                total += prices[i+1]-prices[i]   
        return total
    

if __name__ == '__main__':
    solution_instance = Solution()
    print(solution_instance.maxProfit([7,1,5,3,6,4]))